tests/publiccloud/register_system.pm

```diff
@@ -20,12 +20,14 @@ use publiccloud::utils;
 sub run {
     my ($self, $args) = @_;
 
+    # Preserve args for post_fail_hook
+    $self->{instance} = $args->{my_instance};
+
     select_host_console();    # select console on the host, not the PC instance
 
-    if (is_ondemand) {
-        # on OnDemand image we use `registercloudguest` to register and configure the repositories
-        $args->{my_instance}->retry_ssh_command("sudo registercloudguest", timeout => 420, retry => 3);
```

Comment:
I believe removing this line causes [this failure](http://pdostal-server.suse.cz/tests/13131#step/patch_and_reboot/8)

Reply:
Nope, I was wrong.